### ISSUE TITLE
Memory and space requirements

### ISSUE THREAD

**Issue by zetian1025** [togethercomputer/RedPajama-Data#33]
Hi, I ran the tokenize scripts you provided to tokenize the datasets.
With cl100k_base vocab and tiktokenizer, it result to "can't allocate memory" in my server for every datasets.
Actually the server has over 300G memory, so I'm worry about how much memory and space does it need to tokenize all datasets?

**Comment by mauriceweber** [comment]
We used the GPT-NeoX tokenizer on a machine with 64 cores and 512GB RAM for tokenization (50k vocabulary size), which ran without any memory issues (check out the commands we used [here](https://github.com/togethercomputer/RedPajama-Data/tree/main/tokenization)). Since you use a bigger vocabulary and a different tokenizer, you might need more memory than that. Is there a specific reason that you choose the tiktokenizer over GPT-NeoX?
